### ISSUE TITLE
HTTPS requests over HTTP proxy problem

### ISSUE THREAD

**Issue by zzl221000** [sonic182/aiosonic#393]
**Describe the bug**
Unable to use proxy to access many https sites.

**To Reproduce**
```Python
async with aiosonic.HTTPClient(proxy=Proxy(proxy)) as client:
    resp = await client.get('https://github.com')
    print(resp.status) # 301
    print(resp.headers) # HttpHeaders({'Content-Length': '0', 'Location': 'https://github.com/'})
    resp2 = await client.get('https://github.com/')
    print(resp2.status) # 301
    print(resp2.headers) # HttpHeaders({'Content-Length': '0', 'Location': 'https://github.com/'})
```

**Expected behavior**
Response 200

**Additional context**
After debugging, I found that aiosonic uses the proxy without the CONNECT method to create the tunnel.

**Comment by sonic182** [comment]
Oh thanks!

I have missed that feature as you say, I don't know when I'll find a room for solving it, I hope in a week or two

**Comment by sonic182** [comment]
> **Describe the bug** Unable to use proxy to access many https sites.
> 
> **To Reproduce** Steps to reproduce the behavior:
> 
> ```python
> async with aiosonic.HTTPClient(proxy=Proxy(proxy)) as client:
>     resp = await client.get('https://github.com')
>     print(resp.status) # 301
>     print(resp.headers) # HttpHeaders({'Content-Length': '0', 'Location': 'https://github.com/'})
>     resp2 = await client.get('https://github.com/')
>     print(resp2.status) # 301
>     print(resp2.headers) # HttpHeaders({'Content-Length': '0', 'Location': 'https://github.com/'})
> ```
> 
> **Expected behavior** Response 200
> 
> **Additional context** After debugging, I found that aiosonic uses the proxy without the CONNECT method to create the tunnel.

The fix was not that fast, also I obviously haven't dedicate time to it. In 0.18.0 I advanced a bit in the Connector class, adding some methods for future usage of CONNECT method

Gonna close this issue in favor of https://github.com/sonic182/aiosonic/issues/461 (more or less the same issue)

**Comment by sonic182** [comment]
For now, you can use the Proxy with the code in github at master branch (I did merge this pr https://github.com/sonic182/aiosonic/pull/472)

I'm still missing compatibility for pyhton 3.8.X to 3.10.X

**Comment by sonic182** [comment]
Fixed in https://github.com/sonic182/aiosonic/pull/475, works in latest version (0.18.1)

**Comment by zzl221000** [comment]
> Fixed in #475, works in latest version (0.18.1)已在 #475 中修复，适用于最新版本 （0.18.1）

Thank you, I will try the new version.
